config_buttons_keep.txt
#define DEVICE_NAME "WiT Controller"

// false = Counterclockwise  true = clockwise
#define ENCODER_ROTATION_CLOCKWISE_IS_INCREASE_SPEED true  // changed this to true to have the throttl rotate clockwise

// true = if the locos(s) are stationary, clicking the encoder button will toggle the direction
#define TOGGLE_DIRECTION_ON_ENCODER_BUTTON_PRESSED_WHEN_STATIONAY false

//To enable the function on # you need to change false to true
// #define E_STOP_CURRENT_LOCO true
#define ENCODER_BUTTON_ACTION NEXT_THROTTLE 
#define MAX_THROTTLES 3  // uncomment and increase the number if you need more that two throttles
#define HASH_SHOWS_FUNCTIONS_INSTEAD_OF_KEY_DEFS true

// define what each button will do as direct press (not in a menu)   * and # cannot be remapped
// see static.h or README.md for details on the allowed functions
#define CHOSEN_KEYPAD_0_FUNCTION NEXT_THROTTLE
#define CHOSEN_KEYPAD_1_FUNCTION FUNCTION_1
#define CHOSEN_KEYPAD_2_FUNCTION FUNCTION_2
#define CHOSEN_KEYPAD_3_FUNCTION FUNCTION_3        //changed this to a sound function -- coupler
#define CHOSEN_KEYPAD_4_FUNCTION FUNCTION_9        // changed this to drive hold
#define CHOSEN_KEYPAD_5_FUNCTION FUNCTION_8        // changed this to turn the prime mover (the sound loco) on/off
#define CHOSEN_KEYPAD_6_FUNCTION E_STOP_CURRENT_LOCO  // Might change this to a sound decoder function since I don't use it
#define CHOSEN_KEYPAD_7_FUNCTION DIRECTION_REVERSE // Might change this to a sound function since I have it mapped to an external button
#define CHOSEN_KEYPAD_8_FUNCTION FUNCTION_0   // Might change this to a sound function since I have it mapped to an external button
#define CHOSEN_KEYPAD_9_FUNCTION DIRECTION_FORWARD // Might change this to a sound function since I have it mapped to an external button

// text that will appear when you press #
// if you rearrange the items above, modify this text to suit    * and # cannot be remapped
#define CHOSEN_KEYPAD_0_DISPLAY_NAME "Next Throttle"
#define CHOSEN_KEYPAD_1_DISPLAY_NAME "1 Bell"
#define CHOSEN_KEYPAD_2_DISPLAY_NAME "2 Horn"
#define CHOSEN_KEYPAD_3_DISPLAY_NAME "3 Coupler"
#define CHOSEN_KEYPAD_4_DISPLAY_NAME "4 Drive Hold"
#define CHOSEN_KEYPAD_5_DISPLAY_NAME "5 Prime Mover"
#define CHOSEN_KEYPAD_6_DISPLAY_NAME "Trk On/Off"  // Will relabel this if I change it to a sound function -- See above
#define CHOSEN_KEYPAD_7_DISPLAY_NAME "7 Rev"    // Will relabel this if I change it to a sound function -- See above
#define CHOSEN_KEYPAD_8_DISPLAY_NAME "8 Estop"  // Will relabel this if I change it to a sound function -- See above
#define CHOSEN_KEYPAD_9_DISPLAY_NAME "9 Fwd"    // Will relabel this if I change it to a sound function -- See above

// DCC Addresses below or equal to this number will be treated as short addresses.  
// DCC++EX, NCE, Digitrax use 127, Lenz use 99
#define SHORT_DCC_ADDESS_LIMIT 127  

// speed increase for each click of the encoder 
#define SPEED_STEP 2
#define SPEED_STEP_MULTIPLIER 3  // for 'fast' speed steps
// Additional multiplier.  If the multiplier is enabled from the menu, each rotation of the encoder becomes the speedStep * the AdditionalMultiplier
#define SPEED_STEP_ADDITIONAL_MULTIPLIER 2

// define what each of the optional additional buttons will do
// see static.h or README.md for details on the allowed functions
// all must be included, just set the ones you don't need to FUNCTION_NULL
// the button numbers relate to the GPIO pins 5,15,25,26,27,32,33  -  My external red case buttons starting from the left looking at the throttle
#define CHOSEN_ADDITIONAL_BUTTON_0_FUNCTION FUNCTION_1   // GPIO 5 -Works
#define CHOSEN_ADDITIONAL_BUTTON_1_FUNCTION FUNCTION_2   // GPIO 15 -  Works
#define CHOSEN_ADDITIONAL_BUTTON_2_FUNCTION DIRECTION_REVERSE   // GPIO 25     -  Works
#define CHOSEN_ADDITIONAL_BUTTON_3_FUNCTION FUNCTION_0   // GPIO 26         -- Turns Lights on/off  - Doesn't work  - Uses keypad 0 key – Next Throttle”
#define CHOSEN_ADDITIONAL_BUTTON_4_FUNCTION DIRECTION_FORWARD   // GPIO 27   – Works
#define CHOSEN_ADDITIONAL_BUTTON_5_FUNCTION E_STOP_CURRENT_LOCO  // GPIO 32   - Doesn't work – Uses keypad 8 – Lights
#define CHOSEN_ADDITIONAL_BUTTON_6_FUNCTION POWER_TOGGLE   // GPIO 33  --   - Doesn't work – Uses keypad  6  E-Stop current loco


// uncomment and/or correct one (only) of the #define lines below if you need to override the default .9 inch oLED dispaly settings 
//
// Please select a constructor line for below depending on your display
// U8g2 Constructor List (Frame Buffer)
// The complete list is available here: https://github.com/olikraus/u8g2/wiki/u8g2setupcpp
// Please update the pin numbers according to your setup. Use U8X8_PIN_NONE if the reset pin is not connected
// this is one of the common .9 inch OLED displays
// #define OLED_TYPE U8G2_SSD1306_128X64_NONAME_F_HW_I2C u8g2(U8G2_R0, /* reset=/ U8X8_PIN_NONE, / clock=/ 22, / data=/ 23);
// this is one of the common 1.3 inch OLED displays
#define OLED_TYPE U8G2_SH1106_128X64_NONAME_F_HW_I2C u8g2(U8G2_R0, /* reset=*/ U8X8_PIN_NONE, /* clock=*/ 22, /* data=*/ 23);

#define ADDITIONAL_BUTTON_DEBOUNCE_DELAY 50   // increase if you find the buttons bounce. i.e. activate twice on a single press


config_buttons - problem.txt
#define DEVICE_NAME "WiT Controller"

// false = Counterclockwise  true = clockwise
#define ENCODER_ROTATION_CLOCKWISE_IS_INCREASE_SPEED true  // changed this to true to have the throttl rotate clockwise

// true = if the locos(s) are stationary, clicking the encoder button will toggle the direction
#define TOGGLE_DIRECTION_ON_ENCODER_BUTTON_PRESSED_WHEN_STATIONAY false

//To enable the function on # you need to change false to true
#define E_STOP_CURRENT_LOCO true
#define ENCODER_BUTTON_ACTION NEXT_THROTTLE 
#define MAX_THROTTLES 3  // uncomment and increase the number if you need more that two throttles
#define HASH_SHOWS_FUNCTIONS_INSTEAD_OF_KEY_DEFS true

// define what each button will do as direct press (not in a menu)   * and # cannot be remapped
// see static.h or README.md for details on the allowed functions
#define CHOSEN_KEYPAD_0_FUNCTION NEXT_THROTTLE   // Tried changing this to Next Trottle but isn't working, but I think did with one version of software
#define CHOSEN_KEYPAD_1_FUNCTION FUNCTION_1
#define CHOSEN_KEYPAD_2_FUNCTION FUNCTION_2
#define CHOSEN_KEYPAD_3_FUNCTION FUNCTION_3        //changed this to a sound function -- coupler 3
#define CHOSEN_KEYPAD_4_FUNCTION FUNCTION_9        // changed this to drive hold
#define CHOSEN_KEYPAD_5_FUNCTION FUNCTION_8        // changed this to turn the prime mover (the sound loco) on/off
#define CHOSEN_KEYPAD_6_FUNCTION POWER_TOGGLE  // Might change this to a sound decoder function since I don't use it
#define CHOSEN_KEYPAD_7_FUNCTION DIRECTION_REVERSE // Might change this to a sound function since I have it mapped to an external button
#define CHOSEN_KEYPAD_8_FUNCTION E_STOP_CURRENT_LOCO   // Might change this to a sound function since I have it mapped to an external button
#define CHOSEN_KEYPAD_9_FUNCTION DIRECTION_FORWARD // Might change this to a sound function since I have it mapped to an external button

// text that will appear when you press #
// if you rearrange the items above, modify this text to suit    * and # cannot be remapped
#define CHOSEN_KEYPAD_0_DISPLAY_NAME "Next Throttle"
#define CHOSEN_KEYPAD_1_DISPLAY_NAME "1 Bell"
#define CHOSEN_KEYPAD_2_DISPLAY_NAME "2 Horn"
#define CHOSEN_KEYPAD_3_DISPLAY_NAME "3 Coupler"
#define CHOSEN_KEYPAD_4_DISPLAY_NAME "4 Drive Hold"
#define CHOSEN_KEYPAD_5_DISPLAY_NAME "5 Prime Mover"
#define CHOSEN_KEYPAD_6_DISPLAY_NAME "Trk On/Off"  // Will relabel this if I change it to a sound function -- See above
#define CHOSEN_KEYPAD_7_DISPLAY_NAME "7 Rev"    // Will relabel this if I change it to a sound function -- See above
#define CHOSEN_KEYPAD_8_DISPLAY_NAME "8 Estop"  // Will relabel this if I change it to a sound function -- See above
#define CHOSEN_KEYPAD_9_DISPLAY_NAME "9 Fwd"    // Will relabel this if I change it to a sound function -- See above

// DCC Addresses below or equal to this number will be treated as short addresses.  
// DCC++EX, NCE, Digitrax use 127, Lenz use 99
#define SHORT_DCC_ADDESS_LIMIT 127  

// speed increase for each click of the encoder 
#define SPEED_STEP 2
#define SPEED_STEP_MULTIPLIER 3  // for 'fast' speed steps
// Additional multiplier.  If the multiplier is enabled from the menu, each rotation of the encoder becomes the speedStep * the AdditionalMultiplier
#define SPEED_STEP_ADDITIONAL_MULTIPLIER 2

// define what each of the optional additional buttons will do
// see static.h or README.md for details on the allowed functions
// all must be included, just set the ones you don't need to FUNCTION_NULL
// the button numbers relate to the GPIO pins 5,15,25,26,27,32,33  -  My external red case buttons starting from the left looking at the throttle
#define CHOSEN_ADDITIONAL_BUTTON_0_FUNCTION FUNCTION_1   // GPIO 5   --  Bell On/Off
#define CHOSEN_ADDITIONAL_BUTTON_1_FUNCTION FUNCTION_2   // GPIO 15   -- Horn momentary on
#define CHOSEN_ADDITIONAL_BUTTON_2_FUNCTION DIRECTION_REVERSE   // GPIO 25   -- Changes loco direction to Reverse
#define CHOSEN_ADDITIONAL_BUTTON_3_FUNCTION FUNCTION_0   // GPIO 26         -- Turns Lights on/off
#define CHOSEN_ADDITIONAL_BUTTON_4_FUNCTION DIRECTION_FORWARD   // GPIO 27  -- Changes loco direction to Forward
// #define CHOSEN_ADDITIONAL_BUTTON_5_FUNCTION E_STOP_CURRENT_LOCO  // GPIO 32 -- Emergency stop for loco only
// #define CHOSEN_ADDITIONAL_BUTTON_6_FUNCTION POWER_TOGGLE   // GPIO 33  -- Toggles Track Power on/off/on

// uncomment and/or correct one (only) of the #define lines below if you need to override the default .9 inch oLED dispaly settings 
//
// Please select a constructor line for below depending on your display
// U8g2 Constructor List (Frame Buffer)
// The complete list is available here: https://github.com/olikraus/u8g2/wiki/u8g2setupcpp
// Please update the pin numbers according to your setup. Use U8X8_PIN_NONE if the reset pin is not connected
// this is one of the common .9 inch OLED displays
// #define OLED_TYPE U8G2_SSD1306_128X64_NONAME_F_HW_I2C u8g2(U8G2_R0, /* reset=/ U8X8_PIN_NONE, / clock=/ 22, / data=/ 23);
// this is one of the common 1.3 inch OLED displays
#define OLED_TYPE U8G2_SH1106_128X64_NONAME_F_HW_I2C u8g2(U8G2_R0, /* reset=*/ U8X8_PIN_NONE, /* clock=*/ 22, /* data=*/ 23);

#define ADDITIONAL_BUTTON_DEBOUNCE_DELAY 50   // increase if you find the buttons bounce. i.e. activate twice on a single press


config.h
const char* deviceName = "WiT Controller";

const boolean encoderRotationClockwiseIsIncreaseSpeed = false;
// false = Counterclockwise  true = clockwise

const boolean toggleDirectionOnEncoderButtonPressWhenStationary = true;
// true = if the locos(s) are stationary, clicking the encoder button will toggle the direction

//4x3 keypad
int buttonActions[10] = { E_STOP,           // 0
                         FUNCTION_0,        // 1 - lights
                         FUNCTION_1,        // 2 - bell
                         FUNCTION_2,        // 3 - horn
                         SPEED_MULTIPLIER,  // 4
                         SPEED_UP,          // 5
                         FUNCTION_NULL,     // 6
                         DIRECTION_REVERSE, // 7
                         SPEED_DOWN,        // 8
                         DIRECTION_FORWARD  // 9
};

// text that will appear when you press #
// if you rearrange the items above, modify this text to suit    * and # cannot be remapped
const String directCommandText[4][3] = {
    {"1: F0", "2: F1", "3: F2"},
    {"4: X Spd", "5: V Up", "6: "}, 
    {"7: Rev", "8: V Dwn", "9: Fwd"},
    {"*: Menu", "E Stop", "#: This"}
};

// DCC Addresses below or equal to this number will be treated as short addresses.  
// DCC++EX, NCE, Digitrax use 127, Lenz use 99
#define SHORT_DCC_ADDESS_LIMIT 127  
